applications/FemToDemApplication/custom_elements/generic_small_strain_femdem_element.cpp

```diff
@@ -0,0 +1,1531 @@
+            mThresholds[edge] = element_threhsold;
+        }
+    }
+
+    // IDEM with the edge damages
+    const double damage_element = mDamage;
+    if (norm_2(mDamages) < std::numeric_limits<double>::epsilon()) {
+        for (unsigned int edge = 0; edge < NumberOfEdges; edge++) {
+            mDamages[edge] = damage_element;
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void GenericSmallStrainFemDemElement<3,0>::ComputeEdgeNeighbours(ProcessInfo& rCurrentProcessInfo)
+{
+    this->AuxComputeEdgeNeighbours(rCurrentProcessInfo);
+}
+template<>
+void GenericSmallStrainFemDemElement<3,1>::ComputeEdgeNeighbours(ProcessInfo& rCurrentProcessInfo)
+{
+    this->AuxComputeEdgeNeighbours(rCurrentProcessInfo);
+}
+template<>
+void GenericSmallStrainFemDemElement<3,2>::ComputeEdgeNeighbours(ProcessInfo& rCurrentProcessInfo)
+{
+    this->AuxComputeEdgeNeighbours(rCurrentProcessInfo);
+}
+template<>
+void GenericSmallStrainFemDemElement<3,3>::ComputeEdgeNeighbours(ProcessInfo& rCurrentProcessInfo)
+{
+    this->AuxComputeEdgeNeighbours(rCurrentProcessInfo);
+}
+template<>
+void GenericSmallStrainFemDemElement<3,4>::ComputeEdgeNeighbours(ProcessInfo& rCurrentProcessInfo)
+{
+    this->AuxComputeEdgeNeighbours(rCurrentProcessInfo);
+}
+template<>
+void GenericSmallStrainFemDemElement<3,5>::ComputeEdgeNeighbours(ProcessInfo& rCurrentProcessInfo)
+{
+    this->AuxComputeEdgeNeighbours(rCurrentProcessInfo);
+}
+
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void GenericSmallStrainFemDemElement<2,0>::ComputeEdgeNeighbours(ProcessInfo& rCurrentProcessInfo) {}
+template<> 
+void GenericSmallStrainFemDemElement<2,1>::ComputeEdgeNeighbours(ProcessInfo& rCurrentProcessInfo) {}
+template<> 
+void GenericSmallStrainFemDemElement<2,2>::ComputeEdgeNeighbours(ProcessInfo& rCurrentProcessInfo) {}
+template<> 
+void GenericSmallStrainFemDemElement<2,3>::ComputeEdgeNeighbours(ProcessInfo& rCurrentProcessInfo) {}
+template<> 
+void GenericSmallStrainFemDemElement<2,4>::ComputeEdgeNeighbours(ProcessInfo& rCurrentProcessInfo) {}
+template<> 
+void GenericSmallStrainFemDemElement<2,5>::ComputeEdgeNeighbours(ProcessInfo& rCurrentProcessInfo) {}
+
+/***********************************************************************************/
+/***********************************************************************************/
+template<unsigned int TDim, unsigned int TyieldSurf>
+void GenericSmallStrainFemDemElement<TDim,TyieldSurf>::AuxComputeEdgeNeighbours(
+    ProcessInfo& rCurrentProcessInfo
+    )
+{
+	std::vector<std::vector<Element*>> edge_neighbours_container;
+	Geometry<Node<3>>& r_nodes_current_element = this->GetGeometry();
```

Comment:
Use auto, maybe some day we will remove the template of the node, this will avoid you many problems